packages/web/src/main.tsx

```diff
@@ -1,23 +1,30 @@
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
 import { StrictMode } from 'react';
 import * as ReactDOM from 'react-dom/client';
+import { useTranslation } from 'react-i18next';
 import { RouterProvider } from 'react-router-dom';
-import router from './app/router';
 import './app/i18n';
-import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import router from './app/router';
 import { FlashProvider } from './shared/hooks/flash';
 
+function App() {
+  const { i18n } = useTranslation();
+  document.body.dir = i18n.dir();
```

Comment:
We should only update this when the language changes. We can use `useEffect` to set up event listeners to [i18next events](https://www.i18next.com/overview/api#events)

Reply:
So I should create a local "dir" variable, update it in the event callback, and watch that variable in the useEffect? (I'm still trying to get comfortable with the react way of doing things)

Reply:
Something like this should work:
```ts
useEffect(() => {
  function handler() {
    document.body.dir = i18next.dir()
  }
  i18next.on('initialized', handler)
  i18next.on('languageChanged', handler)
}, [i18next.dir, i18next.on])
```

That will attach event listeners to i18next and only apply changes when the language changes. Both the `dir` and `on` functions should be stable (meaning their reference doesn't change), so this effect will fire only once on first render and attach the listeners.

Reply:
I tried this, but the handler never got called.

Reply:
The problem is that both events trigger before the callback of `useEffect` is called.

Reply:
I think my newest commit fixes it. Instead of using the `initialized` event, I just call the handler directly from the effect.